### PR TITLE
fix(meta): possible race condition in parallel ddl

### PR DESCRIPTION
**Pull request by skyzh**
Signed-off-by: Alex Chi <iskyzh@gmail.com>

## What's changed and what's your intention?

Not sure what makes parallel sqllogictest fail, but some locks could easily solve this. We can investigate later how to remove this lock.

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)

partially fix. https://github.com/singularity-data/risingwave/issues/3051

### CODE REVIEW

File: src/meta/src/rpc/service/ddl_service.rs [hunk 1]
```diff
@@ -228,6 +235,8 @@ where
         &self,
         request: Request<CreateMaterializedViewRequest>,
     ) -> Result<Response<CreateMaterializedViewResponse>, Status> {
+        let _guard = self.ddl_lock.lock().await;
```
Comment (BugenZhao) [inline src/meta/src/rpc/service/ddl_service.rs:238]: A strong taste of C++! 🤣

**Comment by skyzh** [comment]
e2e test didn't get quicker at all, so I'd make this PR on hold.

**Comment by BugenZhao** [comment]
> e2e test didn't get quicker at all, so I'd make this PR on hold.

Is this a performance issue or a correctness issue?

**Comment by TennyZhuang** [comment]
> e2e test didn't get quicker at all, so I'd make this PR on hold.

Why not quicker?

**Comment by skyzh** [comment]
Because we add a big lock on create mv request, which caused everything to run in serial eventually.

**Comment by skyzh** [comment]
> Is this a performance issue or a correctness issue? IIRC the DDL service was designed to be concurrency-safe.

I guess there's correctness issue. e.g. when we create two MV on the same table, it might be possible that we sent two barriers respectively containing info of mv1 and mv2, instead of first mv1, then mv1+mv2.

**Comment by skyzh** [comment]
We don't need quick fix, and maybe we should investigate why it fails in detail.

**Pull request closed without merging**